src/atom.rs

```diff
@@ -67,8 +67,13 @@ impl StringCacheEntry {
 
 impl StringCache {
     fn new() -> StringCache {
+        type T = Option<Box<StringCacheEntry>>;
+        assert!(std::mem::size_of::<T>() == std::mem::size_of::<usize>());
+        let vec = std::mem::ManuallyDrop::new(vec![0_usize; NB_BUCKETS]);
         StringCache {
-            buckets: unsafe { mem::zeroed() },
+            buckets: unsafe {
+                Box::from_raw(vec.as_ptr() as *mut [T; NB_BUCKETS])
```

Comment:
Why not just `vec![None; NB_BUCKETS].into_boxed_slice()` anyway?

Reply:
The bound check when indexing with a masked hash can likely be optimized away if the field is a fixed-size array, maybe not if it’s a slice. Though I haven’t verified either.

https://github.com/servo/string-cache/blob/51b58ce6f94c2b245410bfd70c5d6e1fb7b2a7e0/src/atom.rs#L76-L79

Reply:
Oh I didn't notice the array. My issue wasn't with that part, but with the `0usize` becoming `None` and the size check, I don't think we need that. You can just do `vec![None; NB_BUCKETS]` to create `vec`.

Reply:
I tried `vec![None; NB_BUCKETS]`, it requires `StringCacheEntry: Clone` and deriving that doesn’t work because `AtomicIsize: !Clone`

Reply:
Oh I see!

Reply:
(I even made https://github.com/rust-lang/rust/pull/64891 in anticipation of this…)